### PR TITLE
[FIX] stock: fix recursion stop check

### PR DESCRIPTION
**Pull request by aj-fuentes**
Detected during upgrades from 13.0 to 14.0:
```
  ...
  File "/home/odoo/src/odoo/14.0/addons/stock/models/product.py", line 563, in _get_rules_from_location
    return self._get_rules_from_location(rule.location_src_id, seen_rules=seen_rules | rule)
  [Previous line repeated 942 more times]
  File "/home/odoo/src/odoo/14.0/addons/stock/models/product.py", line 556, in _get_rules_from_location
    'warehouse_id': location.get_warehouse()
  File "/home/odoo/src/odoo/14.0/addons/stock/models/stock_rule.py", line 463, in _get_rule
    result = self._search_rule(values.get('route_ids', False), product_id, values.get('warehouse_id', False), domain)
  File "/home/odoo/src/odoo/14.0/addons/stock/models/stock_rule.py", line 447, in _search_rule
    res = Rule.search(expression.AND([[('route_id', 'in', product_routes.ids)], domain]), order='route_sequence, sequence', limit=1)
  File "/home/odoo/src/odoo/14.0/odoo/models.py", line 1708, in search
    res = self._search(args, offset=offset, limit=limit, order=order, count=count)
  File "/home/odoo/src/odoo/14.0/odoo/models.py", line 4494, in _search
    query = self._where_calc(args)
  File "/home/odoo/src/odoo/14.0/odoo/models.py", line 4250, in _where_calc
    return expression.expression(domain, self).query
  File "/home/odoo/src/odoo/14.0/odoo/osv/expression.py", line 430, in __init__
    self._unaccent = get_unaccent_wrapper(model._cr)
  File "/home/odoo/src/odoo/14.0/odoo/osv/expression.py", line 404, in get_unaccent_wrapper
    if odoo.registry(cr.dbname).has_unaccent:
  File "/home/odoo/src/odoo/14.0/odoo/__init__.py", line 104, in registry
    return modules.registry.Registry(database_name)
  File "/home/odoo/src/odoo/14.0/odoo/modules/registry.py", line 63, in __new__
    return cls.registries[db_name]
  File "/home/odoo/src/odoo/14.0/odoo/tools/func.py", line 71, in wrapper
    return func(self, *args, **kwargs)
RecursionError: maximum recursion depth exceeded
```

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/product.py [hunk 1]
```diff
@@ -555,7 +555,7 @@ def _get_rules_from_location(self, location, route_ids=False, seen_rules=False):
             'route_ids': route_ids,
             'warehouse_id': location.get_warehouse()
         })
-        if not rule:
+        if not rule or rule <= seen_rules:
```
Comment (KangOl) [inline addons/stock/models/product.py:558]:
```suggestion
        if not rule or rule in seen_rules:
```

**Review by KangOl — CHANGES_REQUESTED** [review]
Can you easily add a test case?

File: addons/stock/models/product.py (hunk 1 above)
Comment (aj-fuentes) [reply addons/stock/models/product.py:558]: I'll try.

**Comment by aj-fuentes** [comment]
> Can you easily add a test case?

I'll try.

**Pull request closed without merging**